Admin/Admin.php

```diff
@@ -2540,8 +2546,12 @@ public function id($entity)
     /**
      * {@inheritdoc}
      */
-    public function setValidator(ValidatorInterface $validator)
+    public function setValidator($validator)
```

Comment:
Yeah : you could have an object, implementing `ValidatorInterface` that would hold a reference to the real validator. A bit ugly, but could work.